### ISSUE TITLE
Timers accumulate

### ISSUE THREAD

**Issue by Dbzman** [seaside1/jrule#21]
I have a rule which is triggered by 3 cron entries to check things 3 times a day:
```
public class Abfalltermine extends BaseRule {
    @JRuleName("Abfalltermine")
    @JRuleWhen(hours = 19, minutes = 0)
    @JRuleWhen(hours = 20, minutes = 0)
    @JRuleWhen(hours = 22, minutes = 0)
    public void abfalltermine(JRuleEvent event) {
   ...
   }
}
```
It seems that the timers accumulate with every execution. There are +3 entries every time this gets executed. After several days there's a lot in the logs.
At 19:00 today
```
2022-01-19 19:00:00.187 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 19:00:00.195 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 19 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 19:00:00.201 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 19:00:00.204 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 19 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 19:00:00.210 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 19:00:00.213 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 19 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 19:00:00.219 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 19:00:00.222 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 19 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 19:00:00.228 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 19:00:00.231 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 19 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 19:00:00.237 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 19:00:00.240 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 19 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 19:00:00.246 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 19:00:00.249 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 19 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 19:00:00.252 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 19:00:00.254 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 19 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 19:00:00.257 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 19:00:00.259 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 19 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 19:00:00.262 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 19:00:00.264 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 19 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 19:00:00.268 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 19:00:00.270 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 19 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 19:00:00.273 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 19:00:00.275 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 19 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 19:00:00.278 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 19:00:00.280 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 19 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 19:00:00.283 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 19:00:00.285 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 19 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 19:00:00.288 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 19:00:00.290 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 19 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 19:00:00.294 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 19:00:00.296 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 19 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 19:00:00.299 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 19:00:00.301 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 19 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 19:00:00.304 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 19:00:00.306 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 19 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 19:00:00.309 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 19:00:00.311 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 19 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 19:00:00.314 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 19:00:00.316 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 19 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 19:00:00.319 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 19:00:00.321 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 19 hours: 0 minutes: -1 seconds:  cron: null
```

At 20:00 today
```
2022-01-19 20:00:00.694 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 20:00:00.697 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 20 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 20:00:00.701 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 20:00:00.703 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 20 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 20:00:00.706 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 20:00:00.709 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 20 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 20:00:00.712 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 20:00:00.714 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 20 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 20:00:00.718 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 20:00:00.721 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 20 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 20:00:00.724 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 20:00:00.758 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 20 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 20:00:00.777 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 20:00:00.782 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 20 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 20:00:00.785 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 20:00:00.788 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 20 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 20:00:00.791 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 20:00:00.793 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 20 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 20:00:00.796 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 20:00:00.799 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 20 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 20:00:00.802 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 20:00:00.805 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 20 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 20:00:00.809 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 20:00:00.811 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 20 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 20:00:00.815 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 20:00:00.817 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 20 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 20:00:00.820 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 20:00:00.822 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 20 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 20:00:00.825 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 20:00:00.827 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 20 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 20:00:00.830 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 20:00:00.832 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 20 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 20:00:00.835 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 20:00:00.837 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 20 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 20:00:00.839 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 20:00:00.841 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 20 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 20:00:00.844 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 20:00:00.847 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 20 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 20:00:00.850 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 20:00:00.853 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 20 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 20:00:00.855 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 20:00:00.858 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 20 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 20:00:00.861 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 20:00:00.864 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 20 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 20:00:00.867 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 20:00:00.869 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 20 hours: 0 minutes: -1 seconds:  cron: null
2022-01-19 20:00:00.872 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Timer has finished
2022-01-19 20:00:00.875 [INFO ] [on.jrule.internal.engine.JRuleEngine] - [Abfalltermine] Scheduling timer for rule: 20 hours: 0 minutes: -1 seconds:  cron: null
```

**Comment by seime** [comment]
I've seen the same. Workaround: cron expressions seems to execute correctly

**Comment by Dbzman** [comment]
@seime Thanks a lot. The workaround seems to work fine.

**Comment by seaside1** [comment]
Might be fixed with seimes fix. I'll see if this is reproducible now.

**Comment by seaside1** [comment]
I've tested this seems to work fine now with Seimes fix.
